src/main/java/org/broadinstitute/hellbender/tools/reference/ReferenceSequenceTable.java

```diff
@@ -0,0 +1,437 @@
+                md5 = ReferenceUtils.calculateMD5(referencePath, referenceInterval);
+                if(md5FromDict != null && !md5FromDict.equals(md5)){
+                    logger.warn(String.format("MD5 Mismatch for sequence %s. Found '%s', but calculated '%s'. Sequence dictionary may be invalid.", record.getSequenceName(), md5FromDict, md5));
+                }
+                break;
+        }
+        return md5;
+    }
+
+    /**
+     * Generate ReferencePairs for pairwise comparison of all references present in the table
+     *
+     * @return the list of ReferencePairs for every pair of references
+     */
+    public List<ReferencePair> generateReferencePairs(){
+        List<ReferencePair> referencePairs = new ArrayList<>();
+        for(int i = 0; i < references.size(); i++){
+            for(int j = i + 1; j < references.size(); j++){
+                referencePairs.add(new ReferencePair(this, references.get(i), references.get(j)));
+            }
+        }
+        return referencePairs;
+    }
+
+    /**
+     * Analyze the table by doing a pairwise comparison for all table references. Generates all ReferencePairs, then
+     * analyzes each pair. First, traverses the table by MD5 and checks for any discrepancies across the column values
+     * that violate an exact match or a difference in sequence name. Then traverses by sequence name and checks
+     * each sequence name across the references to determine if the sequence name is present in both but corresponding to
+     * different sequences or missing in one, and if so, if it constitutes a superset/subset case.
+     *
+     * Each ReferencePair is analyzed as above and assigned an analysis as a set of the following statuses:
+     *      EXACT_MATCH,
+     *      DIFFER_IN_SEQUENCE_NAMES,
+     *      DIFFER_IN_SEQUENCE,
+     *      DIFFER_IN_SEQUENCES_PRESENT,
+     *      SUPERSET,
+     *      SUBSET
+     *
+     * @return list of ReferencePairs with updated status sets
+     */
+    public List<ReferencePair> analyzeTable(){
+        checkTableBuildStatus();
+
+        List<ReferencePair> refPairs = generateReferencePairs();
+
+        for(TableRow row : tableByMD5.values()) {
+            for(ReferencePair pair : refPairs) {
+                int ref1Index = pair.getRef1ColumnIndex();
+                int ref2Index = pair.getRef2ColumnIndex();
+                TableEntry[] entries = row.getEntries();
+
+                TableEntry ref1Value = entries[ref1Index];
+                TableEntry ref2Value = entries[ref2Index];
+                if (!ref1Value.getColumnValue().equals(ref2Value.getColumnValue())) {
+                    pair.removeStatus(ReferencePair.Status.EXACT_MATCH);
+                }
+                if(!ref1Value.getColumnValue().equals(ref2Value.getColumnValue()) &&
+                        !(ref1Value.getColumnValue().equals(MISSING_ENTRY_DISPLAY_STRING) || ref2Value.getColumnValue().equals(MISSING_ENTRY_DISPLAY_STRING))){
```

Comment:
Use the `TableEntry.isEmpty()` method whenever you need to check whether an entry is missing. Eg., `ref1Value.isEmpty()` instead of `ref1Value.getColumnValue().equals(MISSING_ENTRY_DISPLAY_STRING)`. Comment also applies to the rest of the method below.